// @dydxprotocol/node-service-lib/build/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.reset = exports.close = exports.authorizationMiddleware = exports.Sequelize = exports.sequelize = exports.Logger = exports.Server = exports.env = undefined;

var _env = require('./helpers/env');

Object.defineProperty(exports, 'env', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_env).default;
  }
});

var _server = require('./server');

Object.defineProperty(exports, 'Server', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_server).default;
  }
});

var _sequelize = require('./helpers/sequelize');

Object.defineProperty(exports, 'sequelize', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_sequelize).default;
  }
});

var _sequelize2 = require('sequelize');

Object.defineProperty(exports, 'Sequelize', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_sequelize2).default;
  }
});

var _authorization = require('./middlewares/authorization');

Object.defineProperty(exports, 'authorizationMiddleware', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_authorization).default;
  }
});

var _validators = require('./helpers/validators');

Object.keys(_validators).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _validators[key];
    }
  });
});

var _errorHandler = require('./helpers/error-handler');

Object.keys(_errorHandler).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _errorHandler[key];
    }
  });
});

var _database = require('./test/database');

Object.defineProperty(exports, 'close', {
  enumerable: true,
  get: function () {
    return _database.close;
  }
});
Object.defineProperty(exports, 'reset', {
  enumerable: true,
  get: function () {
    return _database.reset;
  }
});

var _bignumber = require('bignumber.js');

var _bignumber2 = _interopRequireDefault(_bignumber);

var _logger = require('./helpers/logger');

var _logger2 = _interopRequireDefault(_logger);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Logger = exports.Logger = _logger2.default;


process.on('unhandledRejection', (error, promise) => {
  _logger2.default.error({
    at: 'errorHandler#unhandledRejection',
    message: 'Unhandled Promise Rejection',
    promise,
    error
  });
});

// By default BigNumber's `toString` method converts to exponential notation if the value has
// more then 20 digits. We want to avoid this behavior, so we set EXPONENTIAL_AT to a high number
_bignumber2.default.config({
  EXPONENTIAL_AT: 1000
});
