// @dydxprotocol/node-service-lib/build/helpers/sequelize.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _sequelize = require('sequelize');

var _sequelize2 = _interopRequireDefault(_sequelize);

var _logger = require('./logger');

var _logger2 = _interopRequireDefault(_logger);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const sequelize = new _sequelize2.default(`postgres://${process.env.DB_USERNAME}:${process.env.DB_PASSWORD}` + `@${process.env.DB_HOSTNAME}:${process.env.DB_PORT}/${process.env.DB_NAME}`, {
  logging: false,
  dialect: 'postgres',
  protocol: 'postgres',
  operatorsAliases: _sequelize.Op,
  pool: {
    max: process.env.PG_POOL_MAX,
    min: process.env.PG_POOL_MIN,
    idle: process.env.PG_POOL_IDLE,
    acquire: process.env.PG_POOL_ACQUIRE,
    handleDisconnects: true
  }
});

async function connect() {
  try {
    await sequelize.authenticate();

    _logger2.default.info({
      at: 'sequelize#connect',
      message: 'Connected to postgres'
    });
  } catch (e) {
    if (process.env.NODE_ENV === 'test') {
      throw e;
    }

    _logger2.default.error({
      at: 'sequelize#connect',
      message: 'Failed to connect to postgres',
      error: e
    });
    setTimeout(connect, 5000);
  }
}

sequelize.connect = connect;

exports.default = sequelize;
